### PR TITLE
Use PostCSS loader if a PostCSS config file is present

### PR DESCRIPTION
**Pull request by bjackson**
This will allow users to use PostCSS without modifying their webpack configs.

To enable the postcss-loader, add a `postcss.config.js` file to `web/config`.

As an example, you can use this config file to enable autoprefixer. (Install `yarn workspace web add autoprefixer` to use this).

```js
module.exports = {
    plugins: {
        'autoprefixer': {},
    }
}
```

### CODE REVIEW

File: packages/core/config/webpack.common.js [hunk 1]
```diff
@@ -36,66 +36,87 @@ module.exports = (webpackEnv) => {
...
-              sourceMap: !isEnvProduction,
-            },
-          },
+          styleOrExtractLoader,
+          cssLoader(false, numImportLoadersForSCSS),
+          postCssLoader,
           'sass-loader',
-        ],
+        ].filter(Boolean),
+        sideEffects: true,
```
Comment (peterp) [inline packages/core/config/webpack.common.js:119]: Interesting, what does this do?

Comment (bjackson) [reply packages/core/config/webpack.common.js:119]: I mirrored what create-react-app did with scss/css files. I believe (but not certain) it's for tree-shaking to keep the styles that aren't directly imported in JS from being removed.

https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

Comment (peterp) [reply packages/core/config/webpack.common.js:119]: Super interesting! I didn't know that was possible for css.

**Comment by thedavidprice** [comment]
Thanks for this @bjackson!

 Question: does it look like we need to update this doc based on the changes in this PR? 
https://redwoodjs.com/docs/webpack-configuration#supported-extensions-and-loaders

If so, please loop in @jtoar and me and we'll make sure to update it.

**Comment by bjackson** [comment]
@thedavidprice Yes, we should update the docs to include adding a config to `web/config/postcss.config.js` and installing postcss-loader, instead of altering the postcss-loader options.

**Comment by jtoar** [comment]
Thanks for this @bjackson, doing docs on this now in https://github.com/redwoodjs/redwoodjs.com/pull/219.

**Pull request merged**